ghost/posts-service/lib/PostsService.js

```diff
@@ -61,6 +73,50 @@ class PostsService {
             }
         }
 
+        if (this.isSet('collections') && frame.data.posts[0].collections) {
+            const existingCollections = await this.collectionsService.getCollectionsForPost(frame.options.id);
+            for (const collection of frame.data.posts[0].collections) {
+                let collectionId = null;
+                if (typeof collection === 'string') {
+                    collectionId = collection;
+                }
+                if (typeof collection?.id === 'string') {
+                    collectionId = collection.id;
+                }
+                if (!collectionId) {
+                    continue;
+                }
+                const existingCollection = existingCollections.find(c => c.id === collectionId);
+                if (existingCollection) {
+                    continue;
+                }
+                const found = await this.collectionsService.getById(collectionId);
+                if (!found) {
+                    // TODO: Create collection
```

Comment:
Creating a collection would be a similar logic to what we do for "tags" when we pass in nonexistent ones to the post. In case of collections I don't think we should be having a create logic here, at least for now. I'd change this to "Create collection?" - as we don't know if this is what we want to have here as a feature 😅